src/oss-characteristics/CharacteristicTool.cs

```diff
@@ -310,6 +326,42 @@ public async Task RunAsync(Options options)
                                 var encodedName = Uri.EscapeUriString(target ?? "unknown");
                                 var purl = new PackageURL("generic", encodedName);
                                 AppendOutput(outputBuilder, purl, analysisResults);
+                                finalResults.Add(analysisResults);
+                            }
+                        }
+                        else if (File.Exists(target))
```

Comment:
AppInspector can take a file or archive as argument you don’t need to pre extract and then delete.